packages/vm/src/state/interface.ts

```diff
@@ -21,6 +21,8 @@ export interface StateManager {
   getOriginalContractStorage(address: Address, key: Buffer): Promise<Buffer>
   putContractStorage(address: Address, key: Buffer, value: Buffer): Promise<void>
   clearContractStorage(address: Address): Promise<void>
+  getContractTransientStorage?(address: Address, key: Buffer): Buffer
+  putContractTransientStorage?(address: Address, key: Buffer, value: Buffer): void
   checkpoint(): Promise<void>
```

Comment:
The `StateManager` interface cannot be changed without breaking hardhat, the fork state manager implements this interface. I made the transient storage functions optional to get around this, but that results in some `ts-ignores` throughout this codebase to prevent a "cannot call possibly undefined function" error

https://github.com/NomicFoundation/hardhat/blob/f682724fa16cddfb0a5977b841f274fb743c4657/packages/hardhat-core/src/internal/hardhat-network/provider/fork/ForkStateManager.ts#L45

Reply:
Another option would be to not expose high level setters/getters for transient storage

Reply:
Yes, please definitely not expose something like this in the official interface, we wouldn't want to bloat here for others implementing the interface along doing such experimental integrations like the PR here.

In doubt just do any-casts when using, even if it get's a bit ugly.

Reply:
Will remove from the interface

Reply:
I pushed a fix to remove it from the interface and also added a bit more test coverage